### DISCUSSION TITLE
Fully self contained nocodb instance

### DISCUSSION THREAD

**Discussion by IvanDrag0** [nocodb/nocodb#2703] [Q&A]
Currently, nocodb can build a fully functional self contained Windows executable (using pkg) that can run without having Node or other software installed on the host machine.

I have two questions regarding expanding the capabilities of the self contained executable:

1. Is it possible to package the software in something like Electron or NW.js to have a fully self contained desktop application that has a browser built in (without the need to open the host browser)?
2. When running the executable, nocodb creates a new sqlite database file containing all the settings and initial database (if I don't connect it to any other database) in the same directory. Is it possible to have it use an sqlite database that is embedded within the software and that doesn't create the external file (the file is kept inside the exe)?

**Comment by o1lab** [comment]
Good questions.

1. Packaging to electron can be a bit of work. But the hard part is to sign electron app - these will have to be signed which are a whole lot painful specially microsoft.
2. Not at the moment.

We will think about it in the future.

The reason this not being done is - we do provide APIs as well. So the idea with noco is to run it on servers. Although I see your point. If you can expand on this further - it will be helpful.

**Reply by IvanDrag0** [reply]
I understand that the main use case for nocodb is a server-client architecture where it acts as an "overlay" allowing the user to interface with a data source/database. However, one of the things that I really like about nocodb is its user experience. Its "spreadsheet-like" interface makes the learning curve much smaller since using spreadsheets for data is something most people are familiar with, and it allows me to analyze and present pretty complex data. 

What I wanted to do is use nocodb in a bit of a different use case so that I could utilize its power and user experience in a more portable manner due to the type of networks and systems that the software will be running on (which include small and air-gapped networks). Instead of using it in a server-client overlay, I'd like to use it as a portable self-contained software/executable that has a built-in database (for small to moderate amounts of data). I know that the current architecture of the software is probably a bit of an overkill for that use case but integrating the database into the executable would definitely be a great "start".
